mantidproject/mantid
# Fix crash in MantidPlot's data view

This PR fixes a crash in MantidPlot's data view. The view contains an additional 'X Errors' tab if the underlaying workspace has DX data. If the workspace is replaced by another one without DX, however, the 'X Errors' tab will access nonexistent data resulting in program termination.

**To test:**

One way to get the 'X Errors' tab to show up is to create a single valued workspace:
```python
ws = CreateSingleValuedWorkspace()
```
Open the data view of `ws`, and switch to the 'X Errors' tab. Replace the workspace, e.g.
```python
ws = CreateWorkspace(DataX='0', DataY='0')
```
This should crash MantidPlot before the changes of this PR are applied. With the changes, the 'X Errors' tab should simply disappear.

Fixes #20632. 

**Release Notes** 

A note was added to the 'Workbench' section of UI release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
